### ISSUE TITLE
[6.9.11-55] rounding have changed

### ISSUE THREAD

**Issue by remicollet** [ImageMagick/ImageMagick6#129]
With 16-bit colors, `rgb(0,50%,0)` return 32767 instead of 32768 in previous versions

See https://github.com/Imagick/imagick/issues/373

**Comment by urban-warrior** [comment]
How is the color value produced?  What Q-level are you using?  We get expected results with 6.9.11-55 Q16 from the command-line:

```
$ convert xc:'rgb(0,50%,0)' -compress none  xc.pnm

$ more xc.pnm
P3
1 1
65535
0 32768 0 
```

**Comment by remicollet** [comment]
Really strange...
```
$ convert --version
Version: ImageMagick 6.9.11-55 Q16 x86_64 2021-01-04 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5) 
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype gslib gvc heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png ps raqm raw rsvg tiff webp wmf x xml zlib

$ convert xc:'rgb(0,50%,0)' -compress none  xc.pnm

$ cat xc.pnm 
P3
1 1
65535
0 32767 0 

```

**Comment by remicollet** [comment]
Doesn't seems to be related to GCC version used (same issue on Fedora 31 with GCC 9 and Fedora 33 with GCC 10)

Can it be related to default RPM build options?
`  CFLAGS="${CFLAGS:--O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection}" `

**Comment by urban-warrior** [comment]
Its a floating-point precision issue.  We'll have a patch shortly and a new release within a few days.  Thanks for alerting us to this issue.

**Comment by remicollet** [comment]
Seems fixed in 6.9.11-56
